gm4_pulverizers/data/pulverizers/functions/chance/set_stack_count.mcfunction

```diff
@@ -0,0 +1,12 @@
+#@s = pulverizers containing a recipe results
+#Run from chance/calculate_chance
+
+execute if score result gm4_pv_chance matches ..50 if block ~ ~ ~ dropper{Items:[{Slot:8b,tag:{gm4_pulverizers:{chance:{id:"log"}}}}]} run scoreboard players add chance_stack gm4_stack_size 1
+execute if score result gm4_pv_chance matches ..20 if block ~ ~ ~ dropper{Items:[{Slot:8b,tag:{gm4_pulverizers:{chance:{id:"planks"}}}}]} run scoreboard players add chance_stack gm4_stack_size 1
+execute if score result gm4_pv_chance matches ..10 if block ~ ~ ~ dropper{Items:[{Slot:8b,tag:{gm4_pulverizers:{chance:{id:"blaze_rod"}}}}]} run scoreboard players add chance_stack gm4_stack_size 1
+execute if score result gm4_pv_chance matches ..40 if block ~ ~ ~ dropper{Items:[{Slot:8b,tag:{gm4_pulverizers:{chance:{id:"bone"}}}}]} run scoreboard players add chance_stack gm4_stack_size 1
```

Comment:
Since this function is called as many times as there were items in the first slot, it's probably better to only check the required `result gm4_pv_chance` score one time. Store the `50`, `20`, `10`, etc in a `required gm4_pv_chance` scoreboard inside `apply_chance`.

Another optimization you can do is instead of hard-coding the item type, put the required chance score inside the nbt and copy that to the `required` fakeplayer (see above).